ext/MatrixAlgebraKitCUDAExt/yacusolver.jl

```diff
@@ -242,11 +242,70 @@ for (bname, fname, elty, relty) in
             if jobz == 'V'
                 adjoint!(Vᴴ, Ṽ)
             end
-            return U, S, Vᴴ
+            return S, U, Vᴴ
         end
     end
 end
 
+# Wrapper for randomized SVD
+function Xgesvdr!(A::StridedCuMatrix{T},
+                   S::StridedCuVector=similar(A, real(T), min(size(A)...)),
+                   U::StridedCuMatrix{T}=similar(A, T, size(A, 1), min(size(A)...)),
+                   Vᴴ::StridedCuMatrix{T}=similar(A, T, min(size(A)...), size(A, 2));
+                   k::Int=length(S),
```

Comment:
Would this work? The Xgesvdr docs say that `S` needs to have length `min(size(A)...)`, but also that `k` needs to be (strictly?) less than `min(size(A)...)`. I find the whole CUSOLVER Xgesvdr interface and documentation horrible. In the general explanation part, it mentions $$U$$ as an $$m \times k $$ matrix, and similar for $$\Sigma$$ and $$V$$, and then in the arguments it requires `S`, `U` and `V` to be full size and doesn't mention `k` again:
<img width="1007" height="88" alt="Screenshot 2025-09-03 at 12 24 52" src="https://github.com/user-attachments/assets/9460c60e-3f5a-458f-bdcf-617257f381c1" />

Reply:
This makes me even wonder if `U` really needs to be full size, or only of size $$m \times k$$. Or $$m \times (k+p)$$. I don't understand these interfaces; if `U` is an output variable, why does it need to have a different size than the targetted output. Can't they just use `workspace` for whatever else is needed?

Reply:
It's *very* confusing. I think the idea is that `U` should have size `m x m` **but** only the first `k` columns are actually "valid results"

Reply:
I also suspect it as such, but clearly this part of the documentation has just been copied from other SVD methods without adapting, which makes me wonder whether the suggestion that `U` needs to have size `m x m` for this to work is correct or also just erroneously copied 😄 .